client/src/components/Login.vue

```diff
@@ -53,21 +59,56 @@ export default {
     populateSlackUsers () {
       restFactory.slackUsers((res) => {
         if (res.status === 200) {
-          console.log(res)
+          this.slackUsers = res.data.data
+          this.autocompleteSlackUsers()
         } else {
           global.defaultUserError(res)
         }
       })
     },
+    validateEnteredSlackUser (e) {
+      const userInput = e.target.value
+      const validSlackUsername = this.isValidSlackUsername(userInput)
+      if (validSlackUsername) {
+        this.showingInvalidUsernameError = false
+        const slackUser = this.findSlackUserByUsername(userInput)
+        this.currentSlackUsername = slackUser.username
+        this.currentSlackRealname = slackUser.realname
+        this.currentSlackImage = slackUser.avatar
+      } else {
+        this.currentSlackUsername = ''
+        this.currentSlackRealname = ''
+        this.currentSlackImage = ''
+      }
+    },
+    findSlackUserByUsername (username) {
+      return this.slackUsers.find(user => user.username === username)
+    },
+    isValidSlackUsername (username) {
+      return this.slackUsers.map(user => user.username).includes(username)
+    },
+    autocompleteSlackUsers () {
+      let optionsListString = ''
+      this.slackUsers.forEach((user, index) => {
+        let optionString = '<option value="' + user.username + '" />'
+        optionsListString += optionString
+      })
+      document.getElementById('slack-users-list').innerHTML = optionsListString
+    },
     onSubmit (e) {
-      const formValues = {}
-      const inputs = e.target.elements
-      for (let i = 0; i < inputs.length; i++) {
-        if (inputs[i].value !== 'Submit') {
-          formValues[inputs[i].id] = inputs[i].value
+      if (this.isValidSlackUsername(this.currentSlackUsername)) {
+        this.showingInvalidUsernameError = false
+        const formValues = {}
+        const inputs = e.target.elements
+        for (let i = 0; i < inputs.length; i++) {
+          if (inputs[i].value !== 'Submit') {
+            formValues[inputs[i].id] = inputs[i].value
+          }
         }
+        console.log(formValues)
```

Comment:
do you still need this console log? Just don't want to pollute the logs, I've had a bad experience with that in the past.